tracing,backupccl: log aggregated CapturedStacks during backup

adityamaru opened the pull request:
This change teaches the backup processor to
periodically flush collected AggregatorEvents.
For the time being we only log the CapturedStack structured
events but this sets us up to persist other aggregated
statistics in the future.

Please refer to individual commits for more details.

Release note: None
Informs: https://github.com/cockroachdb/cockroach/issues/100126

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/100316)
<!-- Reviewable:end -->

adityamaru committed:
tracing,backupccl: make CapturedStack an AggregatorEvent

This change moves the TracingAggregator to `pkg/util/tracing`.
It also moves the CapturedStack proto into `pkg/util/tracing`
from `pkg/util/tracing/tracingpb` so that it can implement
the AggregatorEvent interface. `tracingpb` cannot import `tracing`
because of a dependency cycle.

This change also teaches the CapturedStack about the AggregatorEvent
so that CapturedStacks emitted during job execution can be aggreagated
by tracing aggregators.

Release note: None
Informs: #100126

adityamaru committed:
tracing,backupccl: log aggregated CapturedStacks during backup

This change teaches the backup processor to
periodically flush collected AggregatorEvents.
For the time being we only log the CapturedStack structured
events but this sets us up to persist other aggregated
statistics in the future.

Release note: None
Informs: #100126

adityamaru committed:
tracing: rework stack capture to annotate with nodeID and Op

This change reworks how we capture historical stacks
so that the caller can annotate the stacks with the nodeID
and operation name. These two pieces of information are useful
when correlating which slow KV request the stack corresponds to.
When viewing a trace of the span corresponding to the request we
get this information via the span tags, but when the stacks are consumed
outside the context of the tracing span the span tags may or may not
be available for consumption.

This change also adds the UserPriority to the stringer method for
BatchRequests.

Release note: None
rework to annotate store id
